### ISSUE TITLE
"https://127.0.0.1:6443" gets replaced by "http://localhost:8080"

### ISSUE THREAD

**Issue by inteon** [istio/istio#18553]
**Bug description**

Impossible to use a k8s server not on "http://localhost:8080" (ex. "https://127.0.0.1:6443"); configuration gets overwritten by output of depricated function "getDefaultServer()" (see clientcmd).
https://github.com/istio/istio/blob/6c96552ed3954a89989b12f35c2bb91bb5e36b6f/pkg/kube/config.go#L59

This function is being used by https://github.com/kubernetes-sigs/external-dns and causes it to fail connecting to my k8s cluster.

**Expected behavior**
Don't override /.kube/config server value with "http://localhost:8080".

**Steps to reproduce the bug**
`call BuildClientConfig("[home]/.kube/config", "")`

**Version (include the output of `istioctl version --remote` and `kubectl version`)**
latest

**Comment by inteon** [comment]
Removing this line solves the issue for me. https://github.com/istio/istio/blob/6c96552ed3954a89989b12f35c2bb91bb5e36b6f/pkg/kube/config.go#L59

Is this a valid solution? Why was this line added in the first place?

**Comment by howardjohn** [comment]
Introduced in https://github.com/istio/istio/pull/6987

**Comment by ayj** [comment]
I don't remember the exact details, but this was aligning istioctl with kubectl behavior at the time. We switched to an older version of github.com/imdario/mergo to fix a bug in k8s/client-go. Maybe this broken when we moved to a new version?

**Comment by inteon** [comment]
**I found the root cause of the error:**

Our client-go library is pinned to version f68efa97b39e67ee47d4b4683d0037d721f077be (https://github.com/kubernetes/client-go/tree/f68efa97b39e67ee47d4b4683d0037d721f077be).

In client-go version 44530d33a7601800a3eec3ad159a38ac9bfafecc (https://github.com/kubernetes/client-go/commit/44530d33a7601800a3eec3ad159a38ac9bfafecc#diff-b592c01441adb8472d19cb283d369a6bL459-L460) the mergo.Merge(...) function calls were replaced by mergo.MergeWithOverwrite(...) because of a change in mergo.

The mergo version istio is using must be too new for this pinned client-go version.

**This pull-request (https://github.com/istio/istio/pull/18569) should be closed. Removing 'ClusterDefaults' will introduce new errors.**

Instead we should update our client-go version or downgrade our mergo version.
